mcdc/loop.py

```diff
@@ -765,6 +922,107 @@ def davidson(mcdc):
 
 
 @njit(cache=caching)
+def generate_precursor_particle(DNP, particle_idx, seed_work, prog):
```

Comment:
If I understand this we are changing this from wholly within the for loop on 1062 to a separate function

Reply:
You are correct. This is being extracted out, so that it may also be called in `make_work` on line 1104 as well without duplicating the logic. This way, if we change this function, the logic carries over for both CPU and GPU.